pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -19,255 +19,409 @@
+        Map<String, Object> resultForTable = new LinkedHashMap<>();
+        resultForTable.put("tableName", tableNameWithType);
+        resultForTable.put("state", segmentMetadata);
+        resultList.add(Collections.singletonList(resultForTable));
       }
     }
+    if (resultList.isEmpty()) {
+      String errorMessage =
+          "Failed to find segment: " + segmentName + " in table: " + tableName + (tableType != null ? (" of type: "
+              + tableType) : "");
+      throw new ControllerApplicationException(LOGGER, errorMessage, Status.NOT_FOUND);
+    }
+    return resultList;
+  }
 
-    if (result.size() == 0) {
-      String errorMessage = "Failed to find segment: " + segmentName + " in table: " + tableName;
-      if (tableType != null) {
-        errorMessage += " of type: " + tableType;
-      }
-      throw new ControllerApplicationException(LOGGER, errorMessage, Response.Status.NOT_FOUND);
+  @Deprecated
+  @GET
+  @Path("tables/{tableName}/segments/{segmentName}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get the metadata for a segment (deprecated, use 'GET /segments/{tableName}/{segmentName}/metadata' instead)", notes = "Get the metadata for a segment (deprecated, use 'GET /segments/{tableName}/{segmentName}/metadata' instead)")
+  public List<List<Map<String, Object>>> getSegmentMetadataDeprecated2(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "MUST be null") @QueryParam("state") String stateStr,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr) {
+    if (stateStr != null) {
+      throw new WebApplicationException("Cannot toggle segment state", Status.FORBIDDEN);
     }
 
-    return result.toString();
+    return getSegmentMetadataDeprecated1(tableName, segmentName, tableTypeStr);
   }
 
   @POST
-  @Path("tables/{tableName}/segments/{segmentName}/reload")
+  @Path("segments/{tableName}/{segmentName}/reload")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Reloads one segment", notes = "Reloads one segment")
-  public SuccessResponse reloadOneSegment(
+  @ApiOperation(value = "Reload a segment", notes = "Reload a segment")
+  public SuccessResponse reloadSegment(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "Name of segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
-      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr) {
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr) {
```

Comment:
Why do we need table type? Segment name ought to sufficce.

Also, we should verify that there is a segment of that name in the table before we issue a reload. Otherwise we may see other type of exceptions on the server.